exoscale/resource_exoscale_nlb_service.go

```diff
@@ -0,0 +1,430 @@
+			TargetPort:     int64(d.Get("target_port").(int)),
+			Strategy:       d.Get("strategy").(string),
+			Healthcheck: egoscale.NetworkLoadBalancerServiceHealthcheck{
+				Mode:     healthcheck["mode"].(string),
+				Port:     int64(healthcheck["port"].(int)),
+				Interval: interval,
+				Timeout:  timeout,
+				Retries:  int64(healthcheck["retries"].(int)),
+				URI:      healthcheck["uri"].(string),
+			},
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(service.ID)
+
+	log.Printf("[DEBUG] %s: create finished successfully", resourceNlbServiceIDString(d))
+
+	return resourceNlbServiceRead(d, meta)
+}
+
+func resourceNlbServiceRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: beginning read", resourceNlbServiceIDString(d))
+
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutRead))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	service, err := findNlbService(ctx, d, meta)
+	if err != nil {
+		return err
+	}
+
+	if service == nil {
+		return fmt.Errorf("Network Load Balancer Service %q not found", d.Id())
+	}
+
+	log.Printf("[DEBUG] %s: read finished successfully", resourceNlbServiceIDString(d))
+
+	return resourceNlbServiceApply(ctx, client, d, service)
+}
+
+func resourceNlbServiceExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutRead))
+	defer cancel()
+
+	service, err := findNlbService(ctx, d, meta)
+	if err != nil {
+		return false, err
+	}
+
+	if service == nil {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func resourceNlbServiceUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: beginning update", resourceNlbServiceIDString(d))
+
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutUpdate))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	service, err := findNlbService(ctx, d, meta)
+	if err != nil {
+		return err
+	}
+
+	if service == nil {
+		return fmt.Errorf("Network Load Balancer Service %q not found", d.Id())
+	}
+
+	d.Partial(true)
+
+	if d.HasChange("name") {
+		service.Name = d.Get("name").(string)
+		d.SetPartial("name")
+	}
+
+	if d.HasChange("description") {
+		service.Description = d.Get("description").(string)
+		d.SetPartial("description")
+	}
+
+	if d.HasChange("protocol") {
+		service.Protocol = d.Get("protocol").(string)
+		d.SetPartial("protocol")
+	}
+
+	if d.HasChange("strategy") {
+		service.Strategy = d.Get("strategy").(string)
+		d.SetPartial("strategy")
+	}
+
+	if d.HasChange("port") {
+		service.Port = int64(d.Get("port").(int))
+		d.SetPartial("port")
+	}
+
+	if d.HasChange("target_port") {
+		service.TargetPort = int64(d.Get("target_port").(int))
+		d.SetPartial("target_port")
+	}
+
+	if d.HasChange("healthcheck") {
+		raw := d.Get("healthcheck").(*schema.Set)
+		healthcheck := raw.List()[0].(map[string]interface{})
+
+		d.SetPartial("healthcheck")
```

Comment:
No need for partials here, the NLB service is updated atomically.

Reply:
b2a0010